public/app/features/dashboard/components/TransformationsEditor/TransformationEditor.tsx

```diff
@@ -1,29 +1,73 @@
-import React, { useContext } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { mergeMap } from 'rxjs/operators';
 import { css } from 'emotion';
-import { Icon, JSONFormatter, ThemeContext } from '@grafana/ui';
-import { GrafanaTheme, DataFrame } from '@grafana/data';
+import { Icon, JSONFormatter, useStyles } from '@grafana/ui';
+import {
+  DataFrame,
+  DataTransformerConfig,
+  GrafanaTheme,
+  transformDataFrame,
+  TransformerRegistyItem,
+} from '@grafana/data';
 import { selectors } from '@grafana/e2e-selectors';
 
+import { TransformationsEditorTransformation } from './types';
+
 interface TransformationEditorProps {
-  name: string;
-  description?: string;
-  editor?: JSX.Element;
-  input: DataFrame[];
-  output?: DataFrame[];
   debugMode?: boolean;
+  index: number;
+  data: DataFrame[];
+  uiConfig: TransformerRegistyItem<any>;
+  configs: TransformationsEditorTransformation[];
+  onChange: (index: number, config: DataTransformerConfig) => void;
 }
 
-export const TransformationEditor = ({ editor, input, output, debugMode, name }: TransformationEditorProps) => {
-  const theme = useContext(ThemeContext);
-  const styles = getStyles(theme);
+export const TransformationEditor = ({
+  debugMode,
+  index,
+  data,
+  uiConfig,
+  configs,
+  onChange,
+}: TransformationEditorProps) => {
+  const styles = useStyles(getStyles);
+  const [input, setInput] = useState<DataFrame[]>([]);
+  const [output, setOutput] = useState<DataFrame[]>([]);
+  const config = useMemo(() => configs[index], [configs]);
+
+  useEffect(() => {
+    const inputTransforms = configs.slice(0, index).map(t => t.transformation);
+    const outputTransforms = configs.slice(index, index + 1).map(t => t.transformation);
```

Comment:
Previously the output was defined as `configs.slice(index).map(t => t.transformation)`, ie include all transformations starting with the one being debugged. But I changed this because I think it makes more sense to just show the output from the debugged transformation. Easy to change back if I'm missunderstanding this :)

Reply:
Yeah, as long as the input data for the "output" transformation is the result of preceding transformations, which is super elegantly done here https://github.com/grafana/grafana/blob/4c2df34de8fd6f9bbcaa678e5b8ba5d4d0892c49/public/app/features/dashboard/components/TransformationsEditor/TransformationEditor.tsx#L42, this is fine. 

The output should basically show the result of **all preceding transformations & the debugged one**.

Reply:
Perfect, thank you